portal/get_patient_documents.php

```diff
@@ -12,149 +12,134 @@
-    }
-
-    if (!file_exists($directory) || !is_dir($directory)) {
-        return false;
-    } elseif (is_readable($directory)) {
-        $handle = opendir($directory);
-        while (false !== ($item = readdir($handle))) {
-            if ($item != '.' && $item != '..') {
-                $path = $directory . '/' . $item;
-                if (is_dir($path)) {
-                    recursive_remove_directory($path);
-                } else {
-                    unlink($path);
-                }
+?>
+
+<!DOCTYPE html>
+<html>
+<head>
+    <title><?php echo text("Download On File Documents"); ?></title>
+    <?php Header::setupHeader(['no_main-theme', 'portal-theme']); ?>
+    <script>
+        // Function to toggle all checkboxes
+        function toggleCheckboxes(className, sourceCheckbox) {
+            let checkboxes = document.querySelectorAll('.' + className);
+            for (let i = 0; i < checkboxes.length; i++) {
+                checkboxes[i].checked = sourceCheckbox.checked;
             }
         }
 
-        closedir($handle);
-        if ($empty == false) {
-            if (!rmdir($directory)) {
-                return false;
+        // Function to toggle all checkboxes in the form
+        function toggleAllCheckboxes(sourceCheckbox) {
+            let checkboxes = document.querySelectorAll('input[type="checkbox"]');
+            for (let i = 0; i < checkboxes.length; i++) {
+                checkboxes[i].checked = sourceCheckbox.checked;
             }
         }
-    }
-
-    return true;
-}
-
 
-function Zip($source, $destination)
-{
-    if (!extension_loaded('zip') || !file_exists($source)) {
-        return false;
-    }
-
-    $zip = new ZipArchive();
-    if (!$zip->open($destination, ZipArchive::CREATE)) {
-        return false;
-    }
-
-    $source = str_replace('\\', '/', realpath($source));
-    if (is_dir($source) === true) {
-        $files = new RecursiveIteratorIterator(new RecursiveDirectoryIterator($source), RecursiveIteratorIterator::SELF_FIRST);
-        foreach ($files as $file) {
-            if ($file == $source . "/..") {
-                continue;
-            }
+        // Function to normalize category names for use as class names
+        function normalizeClassName(category) {
+            return category.replace(/[^a-zA-Z0-9]/g, '-');
+        }
 
-            $file = str_replace('\\', '/', realpath($file));
-            if (is_dir($file) === true) {
-                $zip->addEmptyDir(str_replace($source . '/', '', $file . '/'));
-            } elseif (is_file($file) === true) {
-                $zip->addFromString(str_replace($source . '/', '', $file), file_get_contents($file));
+        // Function to validate form submission
+        function validateForm(event) {
+            let checkboxes = document.querySelectorAll('input[type="checkbox"]:checked');
+            if (checkboxes.length === 0) {
+                alert("Please select at least one document to download.");
```

Comment:
goodness, i didn't notice several translation issues on this script
would use xlj above